src/Microsoft.AspNet.Razor/Parser/CSharpCodeParser.cs

```diff
@@ -388,7 +388,33 @@ private bool MethodCallOrArrayIndex(AcceptedCharacters acceptedCharacters)
                     }
                     return MethodCallOrArrayIndex(acceptedCharacters);
                 }
-                if (CurrentSymbol.Type == CSharpSymbolType.Dot)
+                if (At(CSharpSymbolType.QuestionMark))
+                {
+                    var next = Lookahead(count: 1);
+
+                    if (next != null)
+                    {
+                        if (next.Type == CSharpSymbolType.Dot)
+                        {
+                            // Accept null propagation dot operator (?.).
```

Comment:
This feature is now consistently called "Null-conditional operators".  See https://github.com/dotnet/roslyn/wiki/New-Language-Features-in-C%23-6